src/utils/filesystem.ts

```diff
@@ -0,0 +1,29 @@
+import { parse, resolve } from 'node:path'
+import { access, constants, mkdir, writeFile } from 'node:fs/promises'
+import { handleFileSystemError } from './error/filesystem-error'
+
+export const saveToFile = async (filePath: string, data: string) => {
+  // Check if the path exists, and create it if it doesn't
+  const resolvedPath = parse(filePath).dir
+  try {
+    await access(resolvedPath, constants.F_OK)
+  }
+  catch {
+    try {
+      await mkdir(resolvedPath, { recursive: true })
+    }
+    catch (mkdirError) {
+      handleFileSystemError('mkdir', mkdirError as Error)
+      return // Exit early if the directory creation fails
+    }
+  }
+
+  try {
+    await writeFile(filePath, data, { mode: 0o600 })
```

Comment:
do we need such a strict permission?

Reply:
Mmm @edodusi I think you are right, I copied and pasted it from the file creation for the `.netrc` since its sensitive data but to the local json files it will be an issue. We don't need it here.

Reply:
@alvarosabu you mistakenly re-added the strict permissions 👀

Reply:
My bad